log/log.go

```diff
@@ -39,10 +41,57 @@ func (f levelFlag) Set(level string) error {
 	return nil
 }
 
+func setFormatterSylog(appname, local string) error {
+	if appname != "" && local != "" {
+		fmter, err := newSyslogger(appname, local, origLogger.Formatter)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "error creating syslog formatter: %v\n", err)
+			origLogger.Errorf("can't connect logger to syslogs: %v", err)
+		} else {
+			origLogger.Formatter = fmter
+		}
+	}
+	return nil
+}
+
+func setJsonFormatter() {
+	origLogger.Formatter = &logrus.JSONFormatter{}
+}
+
+type formatLog struct{}
+
+// String implements flag.Value.
+func (f formatLog) String() string {
+	return origLogger.Level.String()
+}
+
+// Set implements flag.Value.
+func (f formatLog) Set(format string) error {
+	u, err := url.Parse(format)
+	if err != nil {
+		return err
+	}
+	if u.Scheme != "logger" {
+		return fmt.Errorf("Invalid scheme %s", u.Scheme)
+	}
+	jsonq := u.Query().Get("json")
+	if jsonq == "true" {
+		setJsonFormatter()
+	}
+
+	if u.Opaque == "syslog" {
+		appname := u.Query().Get("appname")
+		local := u.Query().Get("local")
+		setFormatterSylog(appname, local)
+	}
+	return nil
+}
+
 func init() {
 	// In order for this flag to take effect, the user of the package must call
 	// flag.Parse() before logging anything.
 	flag.Var(levelFlag{}, "log.level", "Only log messages with the given severity or above. Valid levels: [debug, info, warn, error, fatal].")
+	flag.Var(formatLog{}, "log.format", "If set use a syslog logger or json logging. Example: logger:syslog?appname=bob&local=local7 or logger:stdout@?json=true ")
```

Comment:
So this example shows `local=local7`. The `getLocal` method in the formatter only knows about `local=7` though. Am I missing where you remove the `local` part from `local7`?